docs/tasks/tls/managing-tls-in-a-cluster.md

```diff
@@ -0,0 +1,189 @@
+encoded [pkcs#10](https://tools.ietf.org/html/rfc2986) certification request,
+and `server-key.pem` containing the PEM encoded key to the certificate that is
+still to be created.
+
+### Step 2. Create a Certificate Signing Request object to send to the Kubernetes API
+
+Generate a CSR yaml and send it to the apiserver by running the following command:
+
+```console
+$ cat <<EOF | kubectl create -f -
+apiVersion: certificates.k8s.io/v1beta1
+kind: CertificateSigningRequest
+metadata:
+  name: my-svc.my-namespace
+spec:
+  groups:
+  - system:authenticated
+  request: $(cat server.csr | base64 | tr -d '\n')
+  usages:
+  - digital signature
+  - key encipherment
+  - server auth
+EOF
+```
+
+Notice that the `server.csr` file created in step 1 is base64 encoded and
+stashed in the `.spec.request` field. We are also requesting a certificate
+with the "digital signature", "key encipherment", and "server auth" key
+usages. We support all key usages and extended key usages listed
+[here](https://godoc.org/k8s.io/client-go/pkg/apis/certificates/v1beta1#KeyUsage)
+so you can request client certificates and other certificates using this same
+API.
+
+The CSR should now be visible from the API
+in a Pending state. You can see it by running:
+
+```console
+$ kubectl describe csr my-svc.my-namespace
+Name:                   my-svc.my-namespace
+Labels:                 <none>
+Annotations:            <none>
+CreationTimestamp:      Tue, 21 Mar 2017 07:03:51 -0700
+Requesting User:        yourname@example.com
+Status:                 Pending
+Subject:
+        Common Name:    my-svc.my-namespace.svc.cluster.local
+        Serial Number:
+Subject Alternative Names:
+        DNS Names:      my-svc.my-namespace.svc.cluster.local
+        IP Addresses:   172.168.0.24
+                        10.0.34.2
+Events: <none>
+```
+
+### Step 3. Get the Certificate Signing Request Approved
+
+Approving the certificate signing request is either done by an automated
+approval process or on a one off basis by a cluster administrator. More
+information on what this involves below.
+
+### Step 4. Download the Certificate and Use It
+
+Once the CSR is signed and approved you should see the following:
+
+```console
+$ kubectl get csr
+NAME                  AGE       REQUESTOR               CONDITION
+my-svc.my-namespace   10m       yourname@example.com    Approved,Issued
+```
+
+You can download the issued certificate and save it to a `server.crt` file by
+running the following:
+
+```console
+$ kubectl get csr my-svc.my-namespace -o jsonpath='{.status.certificate}' \
+    | base64 -d > server.crt
+```
+
+Now you can use `server.crt` and `server-key.pem` as the keypair to start your
+HTTPS server.
+
+## Approving Certificate Signing Requests
+
+A Kubernetes administrator (with appropriate permissions) can manually approve
+(or deny) Certificate Signing Requests by using the `kubectl certificate
+approve` and `kubectl certificate deny` commands. However if you intend
```

Comment:
huh, I actually misread that, I thought you were suggesting the 2 steps only applied to automated, since you switched to talking about automated approval. Added suggestion to make that more obvious